addons/mrp_subcontracting/tests/test_subcontracting.py

```diff
@@ -21,6 +21,44 @@ def test_subcontracting_location_1(self):
         self.assertTrue(company2.subcontracting_location_id)
         self.assertTrue(self.env.company.subcontracting_location_id != company2.subcontracting_location_id)
 
+    def test_duplicating_warehouses_recreates_their_routes_and_operation_types(self):
+            """ Duplicating a warehouse should result in the creation of new routes and operation types.
+            Not reusing the existing routes and operation types"""
+            wh_original = self.env['stock.warehouse'].search([], limit=1)
+            wh_copy = wh_original.copy(default={'name': 'Dummy Warehouse (copy)', 'code': 'Dummy'})
+
+            # Check if warehouse routes got RECREATED (instead of reused)
+            route_types = [
+                "route_ids",
+                "pbm_route_id",
+                "subcontracting_route_id",
+                "crossdock_route_id",
+                "reception_route_id",
+                "delivery_route_id"
+            ]
+            for route_type in route_types:
+                original_route_set = getattr(wh_original, route_type)
+                copy_route_set = getattr(wh_copy, route_type)
```

Comment:
the gettattr require a security review, since the fields should exist (and you don't have a fallback anyway getattr would cause an error if the field is not defined) you probably can just use `record[route_type]` and not use attr (hopefully it's not flagged)

same things for the 2 getattr below

Reply:
interesting tip! thanks, lemme try

Reply:
`wh_original[route_type]` doesn't require security review. Switching to it